lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -1263,16 +1263,17 @@ impl STree {
         let chunk_y = (y / self.chunk_size) as i32;
         // Assume that all k nearest neighbors are in these cells
         // it looks like we are rarely wrong once enough pixels are filled in
-        let mut places_to_look = vec![
-            (chunk_x, chunk_y, true),
-            (chunk_x + 1, chunk_y, false),
-            (chunk_x - 1, chunk_y, false),
-            (chunk_x, chunk_y - 1, false),
-            (chunk_x, chunk_y + 1, false),
-            (chunk_x + 1, chunk_y + 1, false),
-            (chunk_x - 1, chunk_y + 1, false),
-            (chunk_x + 1, chunk_y - 1, false),
-            (chunk_x - 1, chunk_y - 1, false),
+        // the stucture of this tuple is (chunk_x, chunk_y, is_center_chunk, closest_x_point, closest_y_point)
+        let places_to_look = vec![
+            (chunk_x, chunk_y, true, 0, 0),
+            (chunk_x + 1, chunk_y, false, (chunk_x + 1) * self.chunk_size as i32, y as i32),
+            (chunk_x - 1, chunk_y, false, chunk_x * self.chunk_size as i32, y as i32),
+            (chunk_x, chunk_y - 1, false, x as i32, chunk_y * self.chunk_size as i32),
+            (chunk_x, chunk_y + 1, false, x as i32, (chunk_y + 1) * self.chunk_size as i32),
+            (chunk_x + 1, chunk_y + 1, false, (chunk_x + 1) * self.chunk_size as i32, (chunk_y + 1) * self.chunk_size as i32),
+            (chunk_x - 1, chunk_y + 1, false, chunk_x * self.chunk_size as i32, (chunk_y + 1) * self.chunk_size as i32),
+            (chunk_x + 1, chunk_y - 1, false, (chunk_x + 1) * self.chunk_size as i32, chunk_y * self.chunk_size as i32),
+            (chunk_x - 1, chunk_y - 1, false, chunk_x * self.chunk_size as i32, chunk_y * self.chunk_size as i32),
         ];
```

Comment:
I would really prefer this to be a `struct` defined inside the the function, tuples are fine for prototyping but end up being really tedious to use over the long term, especially > 2 length.

Reply:
Thanks for the pointer! This code is still pretty prototype-y. I will make a cleanup pass later and can definitely change this.

Reply:
I changed these tuples to a in function struct in my most recent commit